sql/api/src/main/scala/org/apache/spark/sql/streaming/StatefulProcessor.scala

```diff
@@ -27,19 +27,16 @@ import org.apache.spark.annotation.{Evolving, Experimental}
  */
 @Experimental
 @Evolving
-private[sql] trait StatefulProcessor[K, I, O] extends Serializable {
+private[sql] abstract class StatefulProcessor[K, I, O] extends Serializable {
```

Comment:
Do we need to change this ?

Reply:
We want to provide implementation for the getters and setters, so it can no longer be a trait, right?

Reply:
I believe trait can have default/concrete implementation too ?